spec/rspec/mocks/argument_matchers_spec.rb

```diff
@@ -381,16 +381,36 @@ def ==(other)
           a_double.random_call(:a => "a", :b => "b")
         end
 
-        it "matches against a hash submitted by reference and received by value" do
-          opts = {:a => "a", :b => "b"}
-          expect(a_double).to receive(:random_call).with(opts)
-          a_double.random_call(:a => "a", :b => "b")
+        if RUBY_VERSION < '3'
+          it "matches against a hash submitted by reference and received by value in Ruby 2" do
```

Comment:
It seems an issue of the original description, but I think "by value" and "by reference" are not really good terms to describe this (in general, everything is by passed reference in Ruby).
"positional hash" and "keyword arguments" seems far better (which are the terms used in https://www.ruby-lang.org/en/news/2019/12/12/separation-of-positional-and-keyword-arguments-in-ruby-3-0/).

Reply:
I agree. If rspec developers also agree, I'll change the description like this.

> it "matches against a hash submitted as a positional argument and received as a keyword argument in Ruby 2"

Reply:
I'm not aware of the internal implementation, and for me, as a Ruby user, this is more clear. :+1: